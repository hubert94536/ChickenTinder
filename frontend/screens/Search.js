import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { changeFriends, hideError, hideRefresh, showError, showRefresh } from '../redux/Actions.js'
import { connect } from 'react-redux'
<<<<<<< HEAD
import { Dimensions, FlatList, RefreshControl, StyleSheet, Text, View } from 'react-native'
=======
import { Dimensions, FlatList, ImageBackground, StyleSheet, Text } from 'react-native'
>>>>>>> a9ae448c6479c9d09fba04bf2e1739fde5d50041
import { BlurView } from '@react-native-community/blur'
import { SearchBar } from 'react-native-elements'
import PropTypes from 'prop-types'
import accountsApi from '../apis/accountsApi.js'
import Alert from '../modals/Alert.js'
import Card from '../cards/Card.js'
import friendsApi from '../apis/friendsApi.js'
import modalStyles from '../../styles/modalStyles.js'
import screenStyles from '../../styles/screenStyles.js'
import TabBar from '../Nav.js'

// Used to make refreshing indicator appear/disappear
const sleep = (milliseconds) => {
  return new Promise((resolve) => setTimeout(resolve, milliseconds))
}

class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: [],
      friends: [],
      errorAlert: false,
      deleteFriend: false,
      deleteFriendName: '',
      value: '',
      refresh: false,
    }
  }

  componentDidMount() {
    var friendsMap = new Object()
    for (var friend in this.props.friends.friends) {
      friendsMap[this.props.friends.friends[friend].uid] = this.props.friends.friends[friend].status
    }
    this.setState({ friends: friendsMap })
  }

  updateText = async (text) => {
    this.setState({
      value: text,
    })
  }

  searchFilterFunction = async () => {
    clearTimeout(this.timeout) // clears the old timer
    if (!this.state.value) {
      var emptyArray = []
      this.setState({ data: emptyArray })
    } else {
      this.timeout = setTimeout(
        () => {
          if (this.state.value) {
            accountsApi
              .searchUsers(this.state.value)
              .then((res) => {
                var resultUsers = []
                for (var user in res.userList) {
                  var status = 'add'
                  if (res.userList[user].uid in this.state.friends) {
                    status = this.state.friends[res.userList[user].uid]
                  }
                  var person = {
                    name: res.userList[user].name,
                    username: res.userList[user].username,
                    image: res.userList[user].photo,
                    uid: res.userList[user].uid,
                    status: status,
                  }
                  if (person === undefined) {
                    this.setState({ errorAlert: true })
                    return
                  }
                  resultUsers.push(person)
                }
                this.setState({ data: resultUsers })
              })
              .catch(() => {})
          }
        },
        300, //0.3 seconds, adjust as necessary
      )
    }
  }

  async removeRequest(friend, newArr) {
    friendsApi
      .removeFriendship(friend)
      .then(() => {
        this.props.changeFriends(newArr)
        this.setState({ friends: newArr })
      })
      .catch(() => {
        this.setState({ errorAlert: true })
      })
  }

  renderHeader = () => {
    return (
      <SearchBar
        containerStyle={styles.container}
        inputContainerStyle={styles.inputContainer}
        inputStyle={styles.input}
        placeholder="Search for friends"
        lightTheme={true}
        round={true}
        onChangeText={(text) => {
          this.updateText(text).then(() => {
            this.searchFilterFunction()
          })
        }}
        autoCorrect={false}
        value={this.state.value}
      />
    )
  }

  // Called on search-list pulldown refresh
  onRefresh() {
    console.log(this.props.refresh)
    // this.props.showRefresh()
    // sleep(2000).then(this.searchFilterFunction(this.state.value).then(this.props.hideRefresh()))
  }

  render() {
    return (
      <ImageBackground source={require('../assets/backgrounds/Search.png')} style={{ flex: 1 }}>
        <Text style={[screenStyles.icons, styles.title]}>Find friends</Text>
        <FlatList
          data={this.state.data}
          renderItem={({ item }) => (
            <Card
              name={item.name}
              image={item.image}
              uid={item.uid}
              username={item.username}
              currentUser={this.props.username.username}
              total={this.state.data}
              status={item.status}
              key={item.uid}
              press={(uid, newArr) => this.removeRequest(uid, newArr)}
              unfriendAlert={(bool) => this.setState({ deleteFriend: bool })}
            />
          )}
          keyExtractor={(item) => item.username}
          ListHeaderComponent={this.renderHeader}
          onRefresh={() => this.onRefresh()}
          refreshing={this.props.refresh}
        />
        {(this.state.errorAlert || this.state.deleteFriend) && (
          <BlurView
            blurType="dark"
            blurAmount={10}
            reducedTransparencyFallbackColor="black"
            style={modalStyles.blur}
          />
        )}
        {this.state.errorAlert && (
          <Alert
            title="Error, please try again"
            buttonAff="Close"
            height="20%"
            blur
            press={() => this.setState({ errorAlert: false })}
            cancel={() => this.setState({ errorAlert: false })}
          />
        )}
        <TabBar
          goHome={() => this.props.navigation.replace('Home')}
          goSearch={() => {}}
          goNotifs={() => this.props.navigation.replace('Notifications')}
          goProfile={() => this.props.navigation.replace('Profile')}
          cur="Search"
        />
      </ImageBackground>
    )
  }
}

const mapStateToProps = (state) => {
  const { error } = state
  const { refresh } = state
  const { friends } = state
  const { username } = state
  return { error, refresh, friends, username }
}

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      showError,
      hideError,
      showRefresh,
      hideRefresh,
      changeFriends,
    },
    dispatch,
  )

export default connect(mapStateToProps, mapDispatchToProps)(Search)

Search.propTypes = {
  navigation: PropTypes.object,
  // error: PropTypes.bool,
  // refresh: PropTypes.bool,
  // friends: PropTypes.object,
  // username: PropTypes.object,
  showError: PropTypes.func,
  showRefresh: PropTypes.func,
  hideError: PropTypes.func,
  hideRefresh: PropTypes.func,
  changeFriends: PropTypes.func,
}

const styles = StyleSheet.create({
  title: {
    marginTop: '7%',
    textAlign: 'center',
    color: 'white',
  },
  container: {
    backgroundColor: 'white',
    borderBottomColor: 'transparent',
    borderTopColor: 'transparent',
    width: '95%',
    height: Dimensions.get('window').height * 0.08,
    alignSelf: 'center',
    marginTop: '13%',
  },
  inputContainer: {
    height: Dimensions.get('window').height * 0.05,
    width: '90%',
    alignSelf: 'center',
    backgroundColor: '#e7e7e7',
  },
  input: {
    textAlignVertical: 'center',
    fontFamily: screenStyles.medium.fontFamily,
    fontSize: 18,
  },
})
