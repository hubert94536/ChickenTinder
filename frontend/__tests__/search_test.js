import 'react-native'
import { render, fireEvent, cleanup } from '@testing-library/react-native'
import Renderer from 'react-test-renderer'
import Search from '../screens/search'
import MockAsyncStorage from 'mock-async-storage'
import React from 'react'

//import { AsyncStorage } from 'react-native'

const mock = () => {
  const mockImpl = new MockAsyncStorage()
  jest.mock('AsyncStorage', () => mockImpl)
}

mock()

jest.mock('../apis/friendsApi')
jest.mock('../apis/accountsApi.js')

afterEach(cleanup)

test('renders search for nothing correctly', () => {
  const tree = render(<Search />)
  expect(tree.toJSON()).toMatchSnapshot()
})

test('snapshot for search', () => {
  const { getByText, getByPlaceholderText } = render(<Search />)
  // Test typing generates cards
  fireEvent.changeText(getByPlaceholderText('Search for friends'), 'John')
  expect(getByText('j0hn'))
  // Test cards remain after deleting
  fireEvent.changeText(getByPlaceholderText('Search for friends'), '')
  expect(getByText('@j0hn'))
})

test('snapshot for search, no results', () => {
  const tree = render(<Search />)
  const { getByPlaceholderText } = render(<Search />)
  // Test for search that generates nothing
  fireEvent.changeText(getByPlaceholderText('Search for friends'), 'Harbinger')
  expect(tree.toJSON()).toMatchSnapshot()
})

// fail first, accept later

// test('accept friend request calls API + changes button, failing case', () => {

// })

// test('accept friend request calls API + changes button', () => {

// })

// test('delete friend request calls API + removes from view, failing case', () => {

// })

// test('delete friend request calls API + removes from view', () => {

// })

// test('status click if friends, failing case', () => {
//   // test if error alert renders

//   // clicking on cancel closes pop up
// })

// test('status click if friends', () => {
//   // clicking on friend's status creates alert to confirm

//   // clicking on confirm calls Api function + closes pop up
// })
