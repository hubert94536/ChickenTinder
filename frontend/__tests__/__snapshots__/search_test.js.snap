// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search for nothing correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
          "fontSize": 27,
        },
        Object {
          "marginTop": "10%",
          "textAlign": "center",
        },
      ]
    }
  >
    Find friends
  </Text>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 1,
              "borderTopColor": "transparent",
              "borderTopWidth": 1,
              "height": 106.72,
              "padding": 8,
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#e7e7e7",
                  "borderBottomWidth": 0,
                  "borderColor": "#86939e",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 66.7,
                  "minHeight": 30,
                  "overflow": "hidden",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": "90%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 8,
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#86939e",
                              "fontSize": 18,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Search for friends"
                placeholderTextColor="#86939e"
                platform="default"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#86939e",
                    "flex": 1,
                    "fontFamily": "CircularStd-Medium",
                    "fontSize": 18,
                    "marginLeft": 10,
                    "minHeight": 40,
                    "textAlignVertical": "center",
                  }
                }
                testID="searchInput"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#fff2f2",
        "borderRadius": 10,
        "borderWidth": 0,
        "bottom": 0,
        "elevation": 20,
        "flexDirection": "row",
        "height": 93.38000000000001,
        "justifyContent": "space-evenly",
        "marginBottom": "2%",
        "position": "absolute",
        "width": "95%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#F15763",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`snapshot for search, no results 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#F15763",
          "fontFamily": "CircularStd-Medium",
          "fontSize": 27,
        },
        Object {
          "marginTop": "10%",
          "textAlign": "center",
        },
      ]
    }
  >
    Find friends
  </Text>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderBottomColor": "transparent",
              "borderBottomWidth": 1,
              "borderTopColor": "transparent",
              "borderTopWidth": 1,
              "height": 106.72,
              "padding": 8,
              "width": "95%",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "backgroundColor": "#e7e7e7",
                  "borderBottomWidth": 0,
                  "borderColor": "#86939e",
                  "borderRadius": 15,
                  "flexDirection": "row",
                  "height": 66.7,
                  "minHeight": 30,
                  "overflow": "hidden",
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                  ],
                  "width": "90%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginLeft": 8,
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "#86939e",
                              "fontSize": 18,
                            },
                            Object {
                              "backgroundColor": "transparent",
                            },
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="iconIcon"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <TextInput
                allowFontScaling={true}
                autoCorrect={false}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Search for friends"
                placeholderTextColor="#86939e"
                platform="default"
                rejectResponderTermination={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#86939e",
                    "flex": 1,
                    "fontFamily": "CircularStd-Medium",
                    "fontSize": 18,
                    "marginLeft": 10,
                    "minHeight": 40,
                    "textAlignVertical": "center",
                  }
                }
                testID="searchInput"
                underlineColorAndroid="transparent"
                value=""
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": 40,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "marginVertical": 4,
                    "paddingRight": 4,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#fff2f2",
        "borderRadius": 10,
        "borderWidth": 0,
        "bottom": 0,
        "elevation": 20,
        "flexDirection": "row",
        "height": 93.38000000000001,
        "justifyContent": "space-evenly",
        "marginBottom": "2%",
        "position": "absolute",
        "width": "95%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#F15763",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "width": "10%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#8d8d8d",
              "fontSize": 26,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
